Update to Fabric Core 6.0

mikewheaton opened the pull request:
#### Pull request checklist

- [x] Include a change request file if publishing <!-- see notes below -->
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

Updates the Fabric React package and the Fabric Website app to use [Fabric Core 6.0](https://github.com/OfficeDev/office-ui-fabric-core/releases/tag/6.0.0).

Changes include:
- Updating `package.json` in all apps and packages
- Renaming all mixin calls to include the `ms-` prefix
- Updating variable names (added prefixes, changed to kebab-case, etc.)
- Removing any styles that set `font-family`, as this is now inherited from the `ms-Fabric` wrapper
  - This means components now support localized font stacks via the `lang` attribute!
- No longer using the `input-placeholder()` mixin, and instead using `::placeholder` and letting our build system generate the necessary browser prefixes.
- Added a 'component-demo' app to test using a single component and verify that the bundle doesn't contain any unnecessary components or styles.

mikewheaton commented:
@dzearing: I'll take a look at todo-app. Fabric Core 6.0 hasn't actually been released yet, so this branch doesn't yet have a `package.json` update. Right now this is just a test using `npm link` to be sure that the update will go smoothly. More changes coming soon...

mikewheaton commented on common/changes/miwhea-fabric-core-6_2017-02-07-22-38.json:
```diff
@@ -0,0 +1,10 @@
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Update to Fabric Core 6.0.0",
+      "type": "patch"
```
What type is this? It doesn't add functionality or change the API, so I was leaning towards 'patch'. But if developers can use Fabric React to get a copy of Fabric Core, then this is a breaking change since it's a major version of Fabric Core.

battletoilet replied:
I am on the side of this being a breaking change.

mikewheaton replied:
Actually, turns out that if you only have office-ui-fabric-react in your package.json, you won't get a copy of office-ui-fabric-core when you run `npm install`. Think this is because it's under `devDependencies` instead of `dependencies`. So there shouldn't be anyone who's accessing the mixins/variables in Core via React, right?

joschect replied:
It still might cause a lot of problems, especially if a repo consuming fabric is, for whatever reason, still using the old version of Fabric Core. That could lead to css collisions and other issues similar to the ones we saw with the button change.

I could also just be overly cautious now though.

mikewheaton replied:
I'm leaning towards this being a major change. When updating, developers will also need to update their references to Fabric Core 6.0 or there will be breaking changes in the app, such as fonts not being set in components.

battletoilet commented on apps/fabric-website/index.html:
```diff
@@ -65,8 +65,7 @@
         '//cdnjs.cloudflare.com/ajax/libs/react/0.14.7/react-dom.js',
-        'https://static2.sharepointonline.com/files/fabric/fabric-website/fabric-vendor.js',
-        'https://static2.sharepointonline.com/files/fabric/fabric-website/fabric-site.js'
+        'http://localhost:4321/dist/fabric-site.js'
```
this should just be /dist/fabric-site.js

mikewheaton replied:
Fixed, thanks

atneik commented:
Can you also add new icons in core 6.0 to /components/Icon/IconName.ts?

mikewheaton commented:
Good catch, @atneik, I've updated `IconName.ts` for the latest set.

@dzearing, I've added an app under `apps/component-demo` that has a TextField. The app's bundle is 116kb without minification. I don't see any unnecessary styles being included.

mikewheaton commented:
I'm spending too much time trying to keep this branch up to date with master, only to have more changes and merge conflicts introduced. Once this PR is approved I'll deal with the conflicts.

micahgodbolt commented:
Closing this up. Reopen after all of the refactoring is done.
